python/tk_multi_launchapp/launch_apps.py

```diff
@@ -0,0 +1,651 @@
+        meta["app"] = "%s %s" % (self._tk_app.name, self._tk_app.version)
+        meta["launched_engine"] = self._app_engine
+        meta["command"] = command_executed
+        meta["platform"] = sys.platform
+        if ctx.task:
+            meta["task"] = ctx.task["id"]
+        desc =  "%s %s: %s" % (self._tk_app.name, self._tk_app.version, self._menu_name)
+        sgtk.util.create_event_log_entry(
+            self._tk_app.sgtk, ctx, "Toolkit_App_Startup", desc, meta
+        )
+
+    def launch_app_from_entity(self, menu_name, app_engine, app_path, app_args, version=None):
+        """
+        Default method to launch DCC application command based on the current context.
+
+        :param menu_name: Menu name displayed to launch this DCC.
+        :param app_engine: The TK engine associated with the DCC to be launched.
+        :param app_path: Full path to the DCC. May contain environment variables
+                         and/or the locally supported {version}, {v0}, {v1}, ...
+                         variables.
+        :param app_args: Args string to pass to the DCC at launch time.
+        :param version: Specific version of DCC to launch. Used to parse
+                        {version}, {v0}, {v1}, ... information from.
+        """
+        # Verify a Project is defined in the context.
+        if self._tk_app.context.project is None:
+            raise TankError("Your context does not have a project defined. Cannot continue.")
+
+        # Record input DCC info used throughout the launch process
+        self._set_app_info(menu_name, app_engine, app_path, app_args, version)
+
+        # Extract an entity type and id from the context.
+        entity_type = self._tk_app.context.project["type"]
+        entity_id = self._tk_app.context.project["id"]
+        # if there is an entity then that takes precedence
+        if self._tk_app.context.entity:
+            entity_type = self._tk_app.context.entity["type"]
+            entity_id = self._tk_app.context.entity["id"]
+        # and if there is a task that is even better
+        if self._tk_app.context.task:
+            entity_type = self._tk_app.context.task["type"]
+            entity_id = self._tk_app.context.task["id"]
+
+        # Now do the folder creation.  If there is a specific defer keyword, this takes
+        # precedence. Otherwise, use the engine name for the DCC application by default.
+        defer_keyword = self._tk_app.get_setting("defer_keyword") or self._app_engine
+        try:
+            self._tk_app.log_debug("Creating folders for %s %s. Defer keyword: '%s'" %
+                (entity_type, entity_id, defer_keyword)
+            )
+            self._tk_app.sgtk.create_filesystem_structure(entity_type, entity_id, engine=defer_keyword)
+        except sgtk.TankError, err:
+            raise TankError("Could not create folders on disk. Error reported: %s" % err)
+
+        self._launch_app()
+
+    def init_launch_commands(self):
```

Comment:
same comment here: consider `register` rather than `init` as a prefix to hint that the purpose of this method is to register tk commands